examples/control_flow/sequential_executor/src/nat_sequential_executor/register.py

```
@@ -155,9 +156,8 @@ async def analyze_data(processed_data: str) -> str:
             return json.dumps(analysis_results)
 
         except json.JSONDecodeError:
-            # Handle invalid JSON input
-            error_result = {"error": "Invalid input format", "analysis_status": "failed"}
-            return json.dumps(error_result)
+            # Handle invalid JSON input - exit chain early
+            raise SequentialExecutorExit("Invalid input format - cannot proceed")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)